geps/gep-1911.md

```diff
@@ -0,0 +1,260 @@
+
+## Introduction
+
+Since Kubernetes 1.20 the `core/v1.Service` and `core/v1.EndpointSlice` object has a stable `appProtocol` field. It's purpose is to allow end-users to specify an application protocol (L7) for each service port.
+
+Originally the use of this field in the Gateway API was rejected in [GEP-1282](geps/gep-1282.md#non-goals):
+> v1.Service’s appProtocol field is not fit for purpose, because it is defined as accepting values either from the IANA Service Name registry, or domain-prefixed values and we need more flexibility than that. 
+
+Since then a Kubernetes enhancement proposal was created [KEP-3726](https://github.com/kubernetes/enhancements/issues/3726) to repurpose `appProtocol` to include a convention for protocols that are not IANA service names. This would involve prefixing protocol names with `kubernetes.io/*`. 
+
+Note: Kubernetes will automatically create `EndpointSlices` for `Services` that have a selector. [Custom `EndpointSlices`](https://kubernetes.io/docs/concepts/services-networking/service/#custom-endpointslices) can manually be created.
+
+## API Semantics
+
+A Gateway implementation MUST recognize the Kubernetes Standard Application Protocols ([KEP-3726](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/3726-standard-application-protocols)) for specifying the protocol for a backend reference in a Gateway API `*Route` resource
+
+Thus when a `*Route` points to a Kubernetes Service resource the backend protocol for each port can be specified by
+- setting the `appProtocol` field on the Kubernetes `Service`
+- setting the `appProtocol` field on an `Endpoint`/`EndpointSlice` object associated with a Kubernetes `Service`
+
+At the moment there exists two defined constants:
+- `kubernetes.io/h2c` - HTTP/2 over cleartext as described in https://www.rfc-editor.org/rfc/rfc7540
+- `kubernetes.io/raw` - Instructs the implementation to look at the `Service`'s `protocol` field which supports `TCP`, `UDP`, and `SCTP`. Default is `TCP`.
+
+### Multiple Protocols on the same port
+
+Only the Kubernetes `Service` `protocol` field supports multiple protocols on the same port. See the details in [KEP-1435](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/1435-mixed-protocol-lb).
+
+Implementations MUST recognize support for multiple protocols on the same port.
+
+Currently Kubernetes `Service` API does not allow different `appProtocol` values for the same port. At this time there seems to be interest in changing `appProtocol` to be a list in order to faciliate this use-case.
+
+### Default Protocols
+
+If a backend protocol isn't specified an implementation MAY infer the backend protocol through its own means. 
+
+### Unsupported/Incompatible Protocols
+
+If a Route attached to a Gateway is not able to sending traffic to the backend using the specified protocol then it must set it's RouteConditionType `Accepted` to `False`, with a RouteConditionReason `UnsupportedProtocol`
```

Comment:
To clarify, `kubernetes.io/raw` is still not supported. (see [here](https://github.com/kubernetes-sigs/gateway-api/pull/1979#discussion_r1214081474))

I do think it is useful to have one but it would need an small enhancement.

Aside from that, I am not sure I understand the examples.  Why would HTTPRoute not get https, http or `kubernetes.io/h2c` ?

Reply:
I created at able showing some recommended compatability - it does raise more questions